gobblin-runtime/src/main/java/org/apache/gobblin/runtime/StreamModelTaskRunner.java

```diff
@@ -151,9 +152,11 @@ protected void run() throws Exception {
         this.task.configureStreamingFork(fork);
       }
     }
-    new Thread(() -> {
```

Comment:
Just curious: What was the motivation to put this line alone in a new thread? isn't the `connect()` an async call ?

Reply:
Since we are single fork in this case, isForkAsync is false and we don't call observeOn method which cause the connect() method is not an async call and will hang forever until error/finished. It's changed in this PR https://github.com/apache/gobblin/pull/3214/files

Reply:
If that's the case, why not create scheduler anyway (without looking at `isForkAsync`) ? Wrapping thread on top of rxjava doesn't seem to be right.